apps/shared/avifutil.h

```diff
@@ -6,14 +6,18 @@
 
 #include "avif/avif.h"
 
-// The %z format specifier is not available with Visual Studios before 2013 and mingw-w64 toolchains
-// with `__USE_MINGW_ANSI_STDIO` not set to 1. Hence the %I format specifier must be used instead
-// to print out `size_t`. Newer Visual Studios and mingw-w64 toolchains built with the commit
-// mentioned with c99 set as the standard supports the %z specifier properly.
+// The %z format specifier is not available in old version of msvcrt, which is
+// used by Visual Studios before 2013 and usually mingw-w64 toolchains.
+// Hence the %I format specifier must be used instead to print out `size_t`.
+// Visual Studio from 2015 uses UCRT, which supports the %z specifier properly;
+// mingw-w64 toolchains can also be configured to use UCRT.
+// Additionally, with c99 set as the standard mingw-w64 toolchains built with
+// the commit mentioned can patch format functions to support the %z specifier,
+// and this can be detected by `__USE_MINGW_ANSI_STDIO` macro.
 //
 // Related mingw-w64 commit: bfd33f6c0ec5e652cc9911857dd1492ece8d8383
 
-#if (defined(_MSVC) && _MSVC < 1800) || (defined(__USE_MINGW_ANSI_STDIO) && __USE_MINGW_ANSI_STDIO == 0)
+#if !defined(_UCRT) && ((defined(_MSVC) && _MSVC < 1800) || (defined(__USE_MINGW_ANSI_STDIO) && __USE_MINGW_ANSI_STDIO == 0))
```

Comment:
Is it possible to use UCRT with Visual Studio before 2013? Otherwise it can be "simplified" to:

```c++
#if (defined(_MSVC) && _MSVC < 1800) || (defined(__USE_MINGW_ANSI_STDIO) && __USE_MINGW_ANSI_STDIO == 0 && !defined(_UCRT))
```

Reply:
VS <= 2013 never use UCRT and VS >= 2015 always use UCRT.
However, I find this expression easier to understand: 

we want a feature in the C runtime. 
- UCRT definitely provides it.
- msvcr120 (the version of msvcrt Visual Studio 2013 uses) provides it, but no macro can be used to test CRT version so test Visual Studio version instead.
- MinGW sometimes provides it by patch and we can test if the patch is applied.

Reply:
Hi Yuan,

It is reasonable to require Visual Studio 2015 or later today. For example, libaom requires Visual Studio 2017 or later:

https://aomedia.googlesource.com/aom/+/master/README.md#microsoft-visual-studio-builds

Let's drop support for Visual Studio 2013 or older. Could you simplify this expression?

Reply:
Done